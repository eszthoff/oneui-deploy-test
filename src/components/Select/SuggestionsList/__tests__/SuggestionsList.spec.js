import React from 'react';
import toJson from 'enzyme-to-json';
import { render } from 'enzyme';
import { SuggestionsList } from '..';
import { SUGGESTIONS, SUGGESTION_TO_STRING } from '../../../Autosuggest/__mocks__/suggestions';

const noSuggestionsPlaceholder = 'noSuggestionsPlaceholder';
const getItemProps = jest.fn();
const highlightedIndex = 1;
// eslint-disable-next-line react/display-name
const Wrapper = props => (
    <SuggestionsList
        suggestions={SUGGESTIONS}
        suggestionToString={SUGGESTION_TO_STRING}
        noSuggestionsPlaceholder={noSuggestionsPlaceholder}
        getItemProps={getItemProps}
        highlightedIndex={highlightedIndex}
        listOptimizerProps={{
            defaultHeight: 800,
            defaultWidth: 800,
        }}
        {...props}
    />
);

describe('SuggestionsList', () => {
    it('should initially render component correctly', () => {
        const wrapper = render(<Wrapper />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('should render empty component correctly', () => {
        const wrapper = render(<Wrapper suggestions={[]} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
