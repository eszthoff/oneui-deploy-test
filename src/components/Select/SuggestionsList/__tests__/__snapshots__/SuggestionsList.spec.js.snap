// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionsList should initially render component correctly 1`] = `
<div
  style="overflow:visible;height:0;width:0"
>
  <div
    aria-label="grid"
    aria-readonly="true"
    class="ReactVirtualized__Grid ReactVirtualized__List"
    role="grid"
    style="box-sizing:border-box;direction:ltr;height:800px;position:relative;width:800px;-webkit-overflow-scrolling:touch;will-change:transform;overflow-x:hidden;overflow-y:hidden"
    tabindex="0"
  >
    <div
      class="ReactVirtualized__Grid__innerScrollContainer"
      role="rowgroup"
      style="width:auto;height:270px;max-width:800px;max-height:270px;overflow:hidden;pointer-events:;position:relative"
    >
      <li
        class="ListItem ListItem--highlightContext_brand"
        style="height:auto;left:0;position:absolute;top:0;width:100%"
      >
        <div
          class="ListItem__container"
          role="presentation"
        >
          <span>
            Sun
          </span>
        </div>
      </li>
      <li
        class="ListItem ListItem--isHighlighted ListItem--highlightContext_brand"
        style="height:auto;left:0;position:absolute;top:0;width:100%"
      >
        <div
          class="ListItem__container"
          role="presentation"
        >
          <span>
            Moon
          </span>
        </div>
      </li>
      <li
        class="ListItem ListItem--highlightContext_brand"
        style="height:auto;left:0;position:absolute;top:0;width:100%"
      >
        <div
          class="ListItem__container"
          role="presentation"
        >
          <span>
            Mercury
          </span>
        </div>
      </li>
      <li
        class="ListItem ListItem--highlightContext_brand"
        style="height:auto;left:0;position:absolute;top:0;width:100%"
      >
        <div
          class="ListItem__container"
          role="presentation"
        >
          <span>
            Venus
          </span>
        </div>
      </li>
      <li
        class="ListItem ListItem--highlightContext_brand"
        style="height:auto;left:0;position:absolute;top:0;width:100%"
      >
        <div
          class="ListItem__container"
          role="presentation"
        >
          <span>
            Earth
          </span>
        </div>
      </li>
      <li
        class="ListItem ListItem--highlightContext_brand"
        style="height:auto;left:0;position:absolute;top:0;width:100%"
      >
        <div
          class="ListItem__container"
          role="presentation"
        >
          <span>
            Mars
          </span>
        </div>
      </li>
      <li
        class="ListItem ListItem--highlightContext_brand"
        style="height:auto;left:0;position:absolute;top:0;width:100%"
      >
        <div
          class="ListItem__container"
          role="presentation"
        >
          <span>
            Jupiter
          </span>
        </div>
      </li>
      <li
        class="ListItem ListItem--highlightContext_brand"
        style="height:auto;left:0;position:absolute;top:0;width:100%"
      >
        <div
          class="ListItem__container"
          role="presentation"
        >
          <span>
            Saturn
          </span>
        </div>
      </li>
      <li
        class="ListItem ListItem--highlightContext_brand"
        style="height:auto;left:0;position:absolute;top:0;width:100%"
      >
        <div
          class="ListItem__container"
          role="presentation"
        >
          <span>
            Neptun
          </span>
        </div>
      </li>
    </div>
  </div>
</div>
`;

exports[`SuggestionsList should render empty component correctly 1`] = `
<li
  class="ListItem"
>
  <div
    class="ListItem__container"
    role="presentation"
  >
    <p
      class="Text--context_muted"
    >
      noSuggestionsPlaceholder
    </p>
  </div>
</li>
`;
